code/modules/client/client_procs.dm

```diff
@@ -839,3 +840,28 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 /client/proc/AnnouncePR(announcement)
 	if(prefs && prefs.chat_toggles & CHAT_PULLR)
 		to_chat(src, announcement)
+
+/client/proc/show_character_previews(mutable_appearance/MA)
+	LAZYCLEARLIST(char_render_holders)
```

Comment:
You have to reuse the holders if they exist.

Other wise you just stack the new icon over the old one.

Change hair styles a bunch and you'll notice each preview icon is being stacked over the previous one.

If you kept the preview objs and just changed their appearance this wouldn't happen.

Otherwise you'd have to remove them from the screen, but that's a waste when you can rinse and reuse.

Reply:
![image](https://user-images.githubusercontent.com/7069733/50122433-54af8780-0211-11e9-926a-8178e9d12fa1.png)